### PR TITLE
Update EDP requirements

### PR DESCRIPTION
**Pull request by idk-whoami**
* **Addressed Issue(s)**: #7458

* **Server Mode**: re

* **Description of Pull Request**: Corrected script for Poison Bottle items IDs 678 & 22549 + update EDP require 1 list item cost from item_group_db

Thanks to @kaninhot004

### CODE REVIEW

File: db/re/item_db_usable.yml [hunk 1]
```diff
@@ -1823,12 +1823,18 @@ Body:
     Flags:
       BuyingStore: true
     Script: |
-      if (Class == Job_Assassin_Cross || Class == Job_Guillotine_Cross || Class == Job_Guillotine_Cross_T || Class == Job_Shadow_Cross) {
+      switch( Class ) {
```
Comment (Lemongrass3110) [inline db/re/item_db_usable.yml:1826]: Please change this to use eaclass()

File: db/re/item_db_usable.yml [hunk 2]
```diff
@@ -50372,12 +50377,18 @@ Body:
       NoMail: true
       NoAuction: true
     Script: |
-      if (Class == Job_Assassin_Cross || Class == Job_Guillotine_Cross || Class == Job_Guillotine_Cross_T) {
+      switch( Class ) {
```
Comment (Lemongrass3110) [inline db/re/item_db_usable.yml:50380]: Please change this to use eaclass()

**Review by Lemongrass3110 — CHANGES_REQUESTED** [review]

File: db/re/item_db_usable.yml [hunk 3]
```diff
@@ -1823,12 +1823,19 @@ Body:
     Flags:
       BuyingStore: true
     Script: |
-      if (Class == Job_Assassin_Cross || Class == Job_Guillotine_Cross || Class == Job_Guillotine_Cross_T || Class == Job_Shadow_Cross) {
+      switch( eaclass() ) {
+      case EAJ_ASSASSIN_CROSS:
+      case EAJ_GUILLOTINE_CROSS:
+      case EAJ_GUILLOTINE_CROSS_T:
+      case EAJ_BABY_GUILLOTINE_CROSS:
+      case EAJ_SHADOW_CROSS:
```
Comment (Lemongrass3110) [inline db/re/item_db_usable.yml:1831]:
```suggestion
      if ((eaclass()&EAJ_ASSASSIN_CROSS)==EAJ_ASSASSIN_CROSS) {
```

File: db/re/item_db_usable.yml [hunk 4]
```diff
@@ -1823,12 +1823,19 @@ Body:
...
          sc_start SC_DPOISON,60000,0;
          sc_start SC_ASPDPOTION3,60000,9;
-      }
-      else
+         break;
+      Default:
```
Comment (Lemongrass3110) [inline db/re/item_db_usable.yml:1835]:
```suggestion
      }
      else
```

File: db/re/item_db_usable.yml [hunk 5]
```diff
@@ -1823,12 +1823,19 @@ Body:
...
          percentheal -100,-100;
+         break;
+      }
```
Comment (Lemongrass3110) [inline db/re/item_db_usable.yml:1838]:
```suggestion
```

File: db/re/item_db_usable.yml [hunk 6]
```diff
@@ -50372,12 +50378,19 @@ Body:
       NoMail: true
       NoAuction: true
     Script: |
-      if (Class == Job_Assassin_Cross || Class == Job_Guillotine_Cross || Class == Job_Guillotine_Cross_T) {
+      switch( eaclass() ) {
+      case EAJ_ASSASSIN_CROSS:
+      case EAJ_GUILLOTINE_CROSS:
+      case EAJ_GUILLOTINE_CROSS_T:
+      case EAJ_BABY_GUILLOTINE_CROSS:
+      case EAJ_SHADOW_CROSS:
```
Comment (Lemongrass3110) [inline db/re/item_db_usable.yml:50386]:
```suggestion
      if ((eaclass()&EAJ_ASSASSIN_CROSS)==EAJ_ASSASSIN_CROSS) {
```

File: db/re/item_db_usable.yml [hunk 7]
```diff
@@ -50372,12 +50378,19 @@ Body:
...
          sc_start SC_DPOISON,60000,0;
          sc_start SC_ASPDPOTION3,60000,9;
-      }
-      else
+         break;
+      Default:
```
Comment (Lemongrass3110) [inline db/re/item_db_usable.yml:50390]:
```suggestion
      }
      else
```

File: db/re/item_db_usable.yml [hunk 8]
```diff
@@ -50372,12 +50378,19 @@ Body:
...
          percentheal -100,-100;
+         break;
+      }
```
Comment (Lemongrass3110) [inline db/re/item_db_usable.yml:50393]:
```suggestion
```

File: src/map/skill.cpp [hunk 9]
```diff
@@ -17915,6 +17915,18 @@ bool skill_check_condition_castend(struct map_session_data* sd, uint16 skill_id,
 			}
 			break;
 		}
+#ifdef RENEWAL
+		case ASC_EDP:
+			if(sd) {
+				int16 item_edp = itemdb_group.item_exists_pc(sd, IG_EDP);
+				if (item_edp < 0) {
+					clif_skill_fail(sd,skill_id,USESKILL_FAIL_LEVEL,0);
```
[the remark this answers is not in the document]
Comment (idk-whoami) [reply src/map/skill.cpp:17923]: I'm sorry, I don't have an example. Get itemid and amount from itemdb_group.item_exists_pc to use with clif_skill_fail . T_T

**Review by Lemongrass3110 — COMMENTED** [review]
Sorry did not see that my comments were saved in a "review" and not applied directly...

**Review by Lemongrass3110 — APPROVED** [review]

**Review by aleos89 — APPROVED** [review]

**Pull request merged**